internal/affine/colorm.go

```diff
@@ -167,6 +167,158 @@ func (c *ColorM) UnsafeElements() ([]float32, []float32) {
 	return eb, et
 }
 
+func (c *ColorM) det() float32 {
+	if !c.isInited() {
+		return 1
+	}
+
+	m00 := c.body[0]
+	m01 := c.body[1]
+	m02 := c.body[2]
+	m03 := c.body[3]
+	m10 := c.body[4]
+	m11 := c.body[5]
+	m12 := c.body[6]
+	m13 := c.body[7]
+	m20 := c.body[8]
+	m21 := c.body[9]
+	m22 := c.body[10]
+	m23 := c.body[11]
+	m30 := c.body[12]
+	m31 := c.body[13]
+	m32 := c.body[14]
+	m33 := c.body[15]
+
+	b234234 := m22*m33 - m23*m32
```

Comment:
I'm not sure the naming rule here. What does '4' mean?

Reply:
It's from a matrix inversion generator. I'm not sure of the naming pattern, but I don't have an idea for a replacement either. It is used to avoid calculating this more than once (since it's used more than once).

Reply:
> It's from a matrix inversion generator.

What generate did you use?

Reply:
https://github.com/willnode/N-Matrix-Programmer

Reply:
https://github.com/willnode/N-Matrix-Programmer/blob/master/Info/Matrix_4x4.txt '4' doesn't come from this generator.

Reply:
but it's the 5x5:
https://github.com/willnode/N-Matrix-Programmer/blob/master/Info/Matrix_5x5.txt

Reply:
I think we adopted the way to invert the body part (4x4) of the matrix.

Reply:
Or, if the way I suggested and the way of inverting a 5x5 matrix are not so different, that's fine. I'm not sure. What do you think?

Reply:
At least, this function calculates a det of a 4x4 matrix, right?

Reply:
It's the 5x5 without the last column. I removed the multiplications by 0 and 1 (m44).

Reply:
The det is 4x4

Reply:
Let me confirm: I think we have adopted the way of `M_translate^(-1) * M_body^(-1)`, right?

Reply:
You suggested it, but I removed the 5th row and doing a 4x4 inversion, inverting a 4x1 and then multiplying them now doesn't seem to be necessary. It's about the same number of ops combined.

Reply:
Ah OK, acknowledged. As long as the number of ops doesn't change, both are fine. Thank you for explanation!

Then, could you revert det4x4 to det, which is more straight forward? Sorry for the confusion.

Reply:
Are you sure? The det is indeed technically 4x4 because it uses the body (that is 4x4). Let me know

Reply:
Actually this calculates the det of the 5x5 matrix, but with optimization assuming the last column is (0,0,0,0,1), right? The local variable names actually indicates this process as you said. We should rename this function's name back, or, rename the local variables to represent inversion of a 4x4 matrix. What do you think?

Reply:
I renamed the function to avoid further confusion